File: 자바스크립트/큰수만들기.js
function solution(number, k) {
    let Nums = [...number]
    let delNum = 0;
    let stack = [];
    while (Nums.length > 0 && delNum < k) {
        while (stack.length > 0 && stack[stack.length - 1] < Nums[0]) {
            stack.pop()
            delNum++
            if (delNum === k) return stack.concat(Nums).join('')
        }
        stack.push(Nums.shift())
    }

    return stack.slice(0, stack.length - k).join('')
}

function solution2(number, k) {
    var answer = '';
    let stack = [];
    for (const num of number) {
        if (stack.length === 0) {
            stack.push(num)
            continue
        }
        while (stack.length !== 0 && stack[stack.length-1] < num && k > 0) {
            stack.pop()
            k--
        }
        stack.push(num)
    }
    return k === 0 ? stack.join('') : stack.slice(0, -k).join('');
}

File: 큰수만들기.py
from collections import deque


def solution(number, k):
    del_num, stack, number = 0, [], deque(number)

    while number and del_num < k:
        while stack and stack[-1] < number[0]:
            stack.pop()
            del_num += 1
            if del_num == k:
                break
        stack.append(number.popleft())

    stack = stack[:-k] if del_num != k else stack + list(number)

    return ''.join(stack)
